test/testcafe/spec/TerminalView_spec.js

```diff
@@ -123,3 +146,22 @@ async function setup(t) {
       await t.expect(await terminalViewPage.signoutLinkExists()).ok();
     });
 });
+
+// Make sure geolocation is only displayed on OTP Only page when present in response
+[
+  ['Should have entry for geolocation on OTP info page when accessed', terminalReturnOtpOnlyMock, true],
+  ['Should not have entry for geolocation on OTP info page (w/out location) when accessed', terminalReturnOtpOnlyNoLocationMock, false],
+].forEach(([testTitle, mock, expectedValue]) => {
+  test
+    .requestHooks(mock)(testTitle, async t => {
+      const terminalOtpOnlyPage = await setupOtpOnly(t);
+      // Make sure OTP, Browser & OS, App Name are present
+      await t.expect(await terminalOtpOnlyPage.doesOtpEntryExist()).ok();
+      await t.expect(await terminalOtpOnlyPage.doesBrowserOsEntryExist()).ok();
+      await t.expect(await terminalOtpOnlyPage.doesAppEntryExist()).ok();
```

Comment:
Open for debate, but I find value assertions are safer than simply checking if the component is present.

While there is some template logic to **only render** some parts of the view when they exist, providing additional assurance that we aren't passing in invalid values is often beneficial.